pwa-devdocs/src/_drafts/page-builder/overview/index.md

```diff
@@ -1,19 +1,23 @@
+![Page Builder Integration Overview](PageBuilderIntegration.svg)
 
-## Workflow
 
-1. Page Builder content is fetched as part of an entity request to GraphQL (CMS Page, Product Description etc).
-2. `<RichContent />` is employed to determine what type of content this is, using a simple pattern recognition to determine if the content is Page Builder. If the content does not include Page Builder content we opt to render this out as is.
-3. Master format is passed to our dedicated `<PageBuilder />` which in term parses the HTML into objects.
-4. Our `parseStorageHtml` method utilises a `TreeWalker` to iterate over all nodes within the master format, using a pattern to match all content types based on their data attribute.
-5. Each detected content types name is compared to a configuration file of supported content types, if the content type isn't supported we do not render it.
-6. We then pass the master format HTML for the current content type, along with all it's children, to the content types config aggregator.
-7. The config aggregator extracts all required data from the content types format using DOM methods. The config aggregator then returns a simple `{key: value}` object.
-8. An object tree of all content types with their associated data is built and passed to our factory.
-9. The factory requests the component from the config and renders it with the props provided from the config aggregator.
+
+## Detailed flow
+
+As part of the Page Builder / PWA integration, we implemented a system in which we can convert Page Builder's master format into a structured format that React and PWA Studio could understand. We did this on the client side to ensure compatibility with all various hosting methods of Magento Commerce currently available. The flow diagram starts with a customer interacting with the Venia PWA app. 
+
+![Page Builder Integration Details](PageBuilderIntegrationDetails.svg)
+
+1. The Venia app uses a GraphQL query to get CMS content from your Magento store (pages and blocks) and passes the HTML to the `RichContent` component to begin processing.
+2. The `RichContent` component determines if the HTML contains Page Builder content, using simple pattern recognition. If the HTML does not include Page Builder content, it is returned to Venia and rendered out as is. If the HTML does include Page Builder content, the HTML (master format) is passed to the `PageBuilder` component.
+3. The `PageBuilder` component passes the master format HTML to the `parseStorageHTML` function.
+4. The `parseStorageHtml` function uses a `TreeWalker` to parse all nodes within the master format and create an object tree of all the content types with their associated data. The object tree is passed to the `ContentTypeFactory`.
```

Comment:
The ordering here is a little off, `parseStorageHtml` actually calls the config aggregator and passes the configuration (properties) to the factory to then be passed into the component.

Reply:
Got it.